File: Mobile/lib/features/inventory/models/category.dart
class Category {
  final String id;
  final String nameAr;
  final String? nameEn;
  final String? parentCategoryId;
  final bool isActive;

  Category({
    required this.id,
    required this.nameAr,
    this.nameEn,
    this.parentCategoryId,
    required this.isActive,
  });

  factory Category.fromJson(Map<String, dynamic> json) => Category(
        id: json['id'],
        nameAr: json['nameAr'] ?? '',
        nameEn: json['nameEn'],
        parentCategoryId: json['parentCategoryId'],
        isActive: json['isActive'] ?? true,
      );
}

File: Domain/Models/Inventory/Category.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Models.Inventory
{
    public class Category
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required, StringLength(150)]
        public string NameAr { get; set; } = string.Empty;

        [StringLength(150)]
        public string? NameEn { get; set; }

        public Guid? ParentCategoryId { get; set; }
        public Category? ParentCategory { get; set; }

        public ICollection<Category>? SubCategories { get; set; }
        public ICollection<Product>? Products { get; set; }

        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
